cloudant-client/src/main/java/com/cloudant/client/org/lightcouch/internal/CouchDbUtil.java

```diff
@@ -127,12 +127,15 @@ public static String removeExtension(String fileName) {
     }
 
     public static String streamToString(InputStream in) {
-        Scanner s = new Scanner(in, "UTF-8");
-        s.useDelimiter("\\A");
-        String str = s.hasNext() ? s.next() : null;
-        close(in);
-        s.close();// mdb
-        return str;
+        try {
+            Scanner s = new Scanner(in, "UTF-8");
+            s.useDelimiter("\\A");
+            String str = s.hasNext() ? s.next() : null;
+            s.close();// mdb
+            return str;
+        } finally {
+            close(in);
```

Comment:
This applies to the other new `close()`s as well - in some places we do `IOUtils.closeQuietly()` which will swallow the exception. This was probably a convention which built up over time on the `sync-android` library. There are pros and cons about either approach: in the one hand it's generally bad practice to silently swallow exceptions, on the other hand it's not very helpful to throw exceptions out to API users especially when there's little they can do about them.